kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt

```diff
@@ -383,8 +410,9 @@ private fun <T> checkEqualityOfFields(fields: List<KProperty<*>>, value: T, othe
    return fields.mapNotNull {
       val actual = it.getter.call(value)
       val expected = it.getter.call(other)
-      if (actual == expected) null else {
-         "${it.name}: ${actual.show().value} != ${expected.show().value}"
-      }
+
+      val isEqual = eq(actual, expected) == null
```

Comment:
Using `eq` gives us handling of arrays for free.. could it cause other problems? It feels like a nice way to delegate responsibility of equality, IMO

Reply:
Using `eq` will be okay, I don't see any problem using it.